exercises/Fibonacci/typescript/Fibb.ts

```diff
@@ -0,0 +1,31 @@
+var fibCache: number[] = [1, 1]
+
+function fibonacci(m: number): number {
+    if (fibCache.length < m){
+        for (var i = fibCache.length; i <= m; i++) {
+            fibCache[i] = fibCache[i-1] + fibCache[i-2];
+        }
+    }
+    return fibCache[m-1]
+}
+
+// computes the golden ratio
+function computephi(): number {
+    return (Math.sqrt(5) + 1) / 2
+}

```

Comment:
better to make this a value than a function (so that it isn't recomputed each time it is called)